### ISSUE TITLE
node stuck and can not recover by itself

### ISSUE THREAD

**Issue by ianhe8x** [subquery/subql#1796]
### Prerequisites

* [ ] Are you running the latest version(s)?
* [ ] Did you check the [debugging guide](https://doc.subquery.network/academy/tutorials_examples/debug-projects.html#how-to-debug-a-subquery-project)?
* [ ] Did you check the [FAQs](https://doc.subquery.network/faqs/faqs.html) and [Discussions](https://github.com/subquery/subql/discussions)?
* [ ] Are you reporting to the correct repository?
* [ ] Did you search for an existing issue or pull request?

### Description

node stuck and can not recover by itself
it might be related with mmr in db feature

### Details

console outputs:
```
2023-06-12T23:37:03.882Z <benchmark> INFO 0.00 blocks/s. Target height: 15,942,007. Current height: 15,900,864. Estimated time remaining: unknown
2023-06-12T23:37:08.736Z <health> ERROR undefined Error: Indexer is not healthy
2023-06-12T23:37:18.765Z <health> ERROR undefined Error: Indexer is not healthy
2023-06-12T23:37:18.842Z <WorkerBlockDispatcherService> INFO {"threadId":1,"fetchedBlocks":27,"toFetchBlocks":67,"isIndexing":false}
2023-06-12T23:37:18.882Z <benchmark> INFO 0.00 blocks/s. Target height: 15,942,009. Current height: 15,900,864. Estimated time remaining: unknown
2023-06-12T23:37:28.791Z <health> ERROR undefined Error: Indexer is not healthy
2023-06-12T23:37:33.842Z <WorkerBlockDispatcherService> INFO {"threadId":1,"fetchedBlocks":27,"toFetchBlocks":67,"isIndexing":false}
2023-06-12T23:37:33.882Z <benchmark> INFO 0.00 blocks/s. Target height: 15,942,011. Current height: 15,900,864. Estimated time remaining: unknown
2023-06-12T23:37:38.821Z <health> ERROR undefined Error: Indexer is not healthy
2023-06-12T23:37:48.843Z <WorkerBlockDispatcherService> INFO {"threadId":1,"fetchedBlocks":27,"toFetchBlocks":67,"isIndexing":false}
2023-06-12T23:37:48.851Z <health> ERROR undefined Error: Indexer is not healthy
2023-06-12T23:37:48.882Z <benchmark> INFO 0.00 blocks/s. Target height: 15,942,014. Current height: 15,900,864. Estimated time remaining: unknown
2023-06-12T23:37:58.876Z <health> ERROR undefined Error: Indexer is not healthy
2023-06-12T23:38:03.843Z <WorkerBlockDispatcherService> INFO {"threadId":1,"fetchedBlocks":27,"toFetchBlocks":67,"isIndexing":false}
2023-06-12T23:38:03.882Z <benchmark> INFO 0.00 blocks/s. Target height: 15,942,016. Current height: 15,900,864. Estimated time remaining: unknown
2023-06-12T23:38:08.905Z <health> ERROR undefined Error: Indexer is not healthy
2023-06-12T23:38:18.844Z <WorkerBlockDispatcherService> INFO {"threadId":1,"fetchedBlocks":27,"toFetchBlocks":67,"isIndexing":false}
2023-06-12T23:38:18.883Z <benchmark> INFO 0.00 blocks/s. Target height: 15,942,019. Current height: 15,900,864. Estimated time remaining: unknown
2023-06-12T23:38:18.934Z <health> ERROR undefined Error: Indexer is not healthy
2023-06-12T23:38:28.963Z <health> ERROR undefined Error: Indexer is not healthy
2023-06-12T23:38:33.844Z <WorkerBlockDispatcherService> INFO {"threadId":1,"fetchedBlocks":27,"toFetchBlocks":67,"isIndexing":false}
2023-06-12T23:38:33.883Z <benchmark> INFO 0.00 blocks/s. Target height: 15,942,021. Current height: 15,900,864. Estimated time remaining: unknown
```

These details can help to reproduce the environment the issue is occurring

**Local Environment:** [You can get this information from executing `subql version`.]
**Query Version:** [What is the version of the query service?]
**Indexer Version:** 2.5.5
run with mmr in db

**Comment by stwiname** [comment]
How can we reproduce this? Is there a project we can test with

**Comment by jiqiang90** [comment]
> How can we reproduce this? Is there a project we can test with

I think this is same issue as #1795

**Comment by jiqiang90** [comment]
Fixed in #1828
